apps/console/src/hooks/useSubscriptionBanner.ts

```diff
@@ -0,0 +1,30 @@
+import { useGetOrganizationBilling } from '@/lib/graphql-hooks/organization'
+import { parseISO, differenceInDays, isValid } from 'date-fns'
+import { useOrganization } from '@/hooks/useOrganization'
+
+export function useSubscriptionBanner() {
+  const { currentOrgId } = useOrganization()
+  const { data } = useGetOrganizationBilling(currentOrgId)
+
+  const subscription = data?.organization?.orgSubscriptions?.[0]
+  const expiresAt = subscription?.expiresAt
+  const stripeStatus = subscription?.stripeSubscriptionStatus
+  const paymentMethodAdded = subscription?.paymentMethodAdded
+
+  const daysLeft = expiresAt && isValid(parseISO(expiresAt)) ? differenceInDays(parseISO(expiresAt), new Date()) : null
+
+  const isTrial = stripeStatus === 'trialing'
+  const isExpiringSoon = daysLeft !== null && daysLeft <= 7
+
+  let bannerText = ''
+
+  if (isTrial && !paymentMethodAdded && isExpiringSoon) {
+    bannerText = 'Your 30-day free trial ends soon, and there is no payment method on file'
```

Comment:
Lets update to add days left as Matt suggested: 
```suggestion
    bannerText = 'Your trial ends in" + daysLeft ", and there is no payment method on file'
```